### ISSUE TITLE
Fix default route/not found behavior

### ISSUE THREAD

**Issue by TehShrike** [TehShrike/hash-brown-router#11]
The difference between `evaluateCurrent` and `setDefault` is confusing and wrong, and is making life in the abstract-state-router difficult (see https://github.com/TehShrike/abstract-state-router/issues/63).

Right now `evaluateCurrent` checks the current route, and if it is empty, navigates to the default.  If the current route is not empty, then it tries to navigate to it, with the special rule that for that one command, the default route is whatever was passed in to `evaluateCurrent` instead of the usual default route.

Instead of this, `evaluateCurrent` should get the current route, and
- if the route is empty, go to the default
- if the route is not empty, navigate to it, and emit a "not found" event if that url is not matched

Along this vein, I think that `setDefault` isn't the right way to handle invalid urls.  There should probably be an `onNotFound` callback instead.

So to sum up: on `evaluateCurrent`, if there is no url, the default url should be visited.  If a url is visited and has no route, an `onNotFound` callback should be called.

This would be a breaking change.

**Comment by TehShrike** [comment]
I now regret this implementation.  I'm going to flip it over to being an event emitter.
